### PR TITLE
New Code Samples for Netherite

### PR DESCRIPTION
**Pull request by lechnerc77**
This pull request comprises additional code samples (TypeScript and Python) using Netherite in order to make the first steps easy for non-.NET users

- Restructuring of setup
  - Moving of generic installation scripts into a dedicated folder
  - Rewriting of scripts for usage with node and python
  - reorganizing scripts (run and deployment are now contained in the folder for the languages)
- Adding implementations for TypeScript and Python
- Synchronizing implementations to be able to execute the functions:
  - locally without any Azure resource
  - hybrid with the Function running locally and the storage and Event Hub in Azure
  - deployed to Azure

It also contains some changes to the existing setup namely:

- the `settings.ps1` now throws an exception in case that the name is not adopted, stopping the execution
- The .NET Function now only has a POST endpoint and authorization level is function
- The creation of the storage account contains some more parameters that might prevent the execution due to typical security policies in Azure tenants

There will be an additional pull-request to adopt the GitHub pages to the new structure

### CODE REVIEW

File: samples/scripts/settings.ps1 [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+# optionally, customize the following parameters
+# to use different names for resource group, namespace, function app, storage account, and plan
+$groupName = $name
+$nameSpaceName = $name
+$functionAppName = $name
+$storageName = $name
+$planName = $name
+
+if (($name -eq "globally-unique-lowercase-alphanumeric-name-with-no-dashes")) {
```
Comment (sebastianburckhardt) [inline samples/scripts/settings.ps1:23]: I think this test condition should be updated to `($name -eq "testnetheritescripts")` ... unless I am missing something.

File: samples/scripts/create-function-app.ps1 [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/pwsh
+
+# review these parameters before running the script
+param (
+	$Plan = "EP1", 
+	$MinNodes = "1", 
+	$MaxNodes = "20", 
+	$Runtime = "python",
```
[the remark this answers is not in the document]
Comment (sebastianburckhardt) [reply samples/scripts/create-function-app.ps1:8]: Perhaps we could have runtime specific scripts that call this one, e.g. `create-dotnet-function-app.ps1`, `create-python-function-app.ps1`, `create-typescript-function-app.ps1`

File: samples/scripts/request.http [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+### Start Orchestrator Function via HTTP for Python and NodeJS 
+POST http://localhost:7071/api/orchestrators/DurableFunctionsNetheriteOrchestrator
+
+### Start Orchestrator Function via HTPP for .NET Core
```
Comment (sebastianburckhardt) [inline samples/scripts/request.http:5]: HTPP -> HTTP

**Review by sebastianburckhardt — CHANGES_REQUESTED** [review]
This looks great! Thanks a lot for creating these samples, I think they will be very helpful.

I had some minor comments and questions, see review. Also, I would suggest that you include the [README.md](https://github.com/lechnerc77/durablefunctionsnetherite/blob/main/README.md) in the typescript sample, since I found it very helpful.

File: samples/Hello_Netherite_with_DotNetCore/HelloCities.cs [hunk 4]
```diff
@@ -25,7 +25,7 @@ public static class HelloCities
     {
         [FunctionName(nameof(HelloCities))]
         public async static Task<IActionResult> Run(
-            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = "hellocities")] HttpRequest req,
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = "hellocities")] HttpRequest req,
```
[the remark this answers is not in the document]
Comment (lechnerc77) [reply samples/Hello_Netherite_with_DotNetCore/HelloCities.cs:28]: Had the risk in mind, but you are right, the samples in the documentation are anonymous. I changed that for all implementations. I will make a remark in the documentation

File: samples/Hello_Netherite_with_Python/DurableFunctionsNetheriteStarter/function.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{
+  "scriptFile": "__init__.py",
+  "bindings": [
+    {
+      "authLevel": "function",
```
[the remark this answers is not in the document]
Comment (lechnerc77) [reply samples/Hello_Netherite_with_Python/DurableFunctionsNetheriteStarter/function.json:5]: Done

File: samples/scripts/settings.ps1 (hunk 1 above)
Comment (lechnerc77) [reply samples/scripts/settings.ps1:23]: The other way round - did not changed that back to your setup after my last test

File: samples/scripts/request.http (hunk 3 above)
Comment (lechnerc77) [reply samples/scripts/request.http:5]: Done

File: samples/scripts/create-function-app.ps1 (hunk 2 above)
[the remark this answers is not in the document]
Comment (lechnerc77) [reply samples/scripts/create-function-app.ps1:8]: Good point. I created three scripts now referencing the generic one.

**Comment by lechnerc77** [comment]
Concerning the README-file. I will rework the documentation on the samples in the gh-pages branch and will take that into account

**Comment by lechnerc77** [comment]
The documentation is now also updated see pull request <https://github.com/microsoft/durabletask-netherite/pull/73>

**Review by sebastianburckhardt — APPROVED** [review]
Looks good to me! Thanks again.

I haven't tried running the samples on Linux yet, but at least on Windows it all works.

**Comment by sebastianburckhardt** [comment]
@lechnerc77, I am ready to merge this and #73, unless you have some additional changes you want to add first.

**Comment by lechnerc77** [comment]
@sebastianburckhardt : Nothing to add from my side - let the merging begin ;-)

**Pull request merged**